### PR TITLE
Move ghc warning flags from stack.yaml to cabal file

### PR DESCRIPTION
**Pull request by erikd**

### CODE REVIEW

File: shelley/chain-and-ledger/executable-spec/src/EpochBoundary.hs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {-# LANGUAGE TemplateHaskell       #-}
 {-# LANGUAGE TupleSections         #-}
+{-# OPTIONS_GHC -Wno-name-shadowing #-}
```
Comment (mgudemann) [inline shelley/chain-and-ledger/executable-spec/src/EpochBoundary.hs]: closing parentheses could be aligned with the above

File: shelley/chain-and-ledger/executable-spec/src/STS/Utxo.hs [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {-# LANGUAGE EmptyDataDecls #-}
 {-# LANGUAGE TypeFamilies   #-}
+{-# OPTIONS_GHC -Wno-name-shadowing #-}
```
Comment (mgudemann) [inline shelley/chain-and-ledger/executable-spec/src/STS/Utxo.hs]: as above

File: shelley/chain-and-ledger/executable-spec/src/STS/Utxow.hs [hunk 3]
```diff
@@ -2,6 +2,7 @@
 {-# LANGUAGE MultiParamTypeClasses #-}
 {-# LANGUAGE TypeApplications      #-}
 {-# LANGUAGE TypeFamilies          #-}
+{-# OPTIONS_GHC -Wno-name-shadowing #-}
```
Comment (mgudemann) [inline shelley/chain-and-ledger/executable-spec/src/STS/Utxow.hs]: as above

File: shelley/chain-and-ledger/executable-spec/src/EpochBoundary.hs (hunk 1 above)
Comment (erikd) [reply shelley/chain-and-ledger/executable-spec/src/EpochBoundary.hs]: We  are about to add a `stylish-haskell` config that would align that as:
```
{-# LANGUAGE TemplateHaskell #-}
{-# LANGUAGE TupleSections #-}
{-# OPTIONS_GHC -Wno-name-shadowing #-}
```
so I thought it was not worth it.

Comment (erikd) [reply shelley/chain-and-ledger/executable-spec/src/EpochBoundary.hs]: I removed all the `-Wno-name-shadowing` and used `hiding` instead.

**Pull request merged**